ephypype/gather/gather_results.py

```diff
@@ -0,0 +1,266 @@
+            if psd_file:
+                psd_files.append(psd_file[0])
+            if coo_file:
+                channel_coo_files.append(coo_file[0])
+
+    return psd_files, channel_coo_files
+
+
+def get_inverse_files(workflow_path, workflow_name, subject_id, session_id):
+    """
+    Get source reconstruction files.
+
+    Parameters
+    ----------
+       workflow_path : str
+           Path of connectivity workflow
+       workflow_name : str
+           Name of the connectivity workflow
+       subject_id : list of str
+           List of subjects id
+       session_id : list of str
+           List of sessions
+
+    Returns
+    -------
+        time_series_file : list of str
+            List of path of source reconstruction files
+        label_files : list of str
+            List of path of labels pickle file
+    """
+    inverse_pipeline = 'inv_sol_pipeline'
+    inverse_node = 'inv_solution'
+
+    time_series_files = list()
+    label_files = list()
+
+    subjects_folder = '_session_id_{ses}_subject_id_{sbj}'
+    node_path = op.join(workflow_path, workflow_name, inverse_pipeline,
+                        subjects_folder, inverse_node)
+
+    for sbj in subject_id:
+        for ses in session_id:
+            file_path = node_path.format(ses=ses, sbj=sbj)
+            inverse_file = glob.glob(op.join(file_path, '*.npy'))
+            label_file = glob.glob(op.join(file_path, '*.pkl'))
+            if inverse_file:
+                time_series_files.append(inverse_file[0])
+            if label_file:
+                label_files.append(label_file[0])
+
+    return time_series_files, label_files
+
+
+def get_ica_solution(workflow_path, workflow_name, subject_id, session_id):
+    """
+    Get ica solution.
+
+    Parameters
+    ----------
+       workflow_path : str
+           Path of connectivity workflow
+       workflow_name : str
+           Name of the connectivity workflow
+       subject_id : list of str
+           List of subjects id
+       session_id : list of str
+           List of sessions
+
+    Returns
+    -------
+        ica_solution_file : list of str
+            List of path of source reconstruction files
+        label_files : list of str
+            List of path of labels pickle file
+    """
+    ica_pipeline = 'preproc_meeg_pipeline'
+    ica_node = 'ica'
+
+    ica_files = list()
+    raw_files = list()
+
+    subjects_folder = '_session_id_{ses}_subject_id_{sbj}'
+    node_path = op.join(workflow_path, workflow_name, ica_pipeline,
+                        subjects_folder, ica_node)
+
+    for sbj in subject_id:
+        for ses in session_id:
+            file_path = node_path.format(ses=ses, sbj=sbj)
+            ica_file = glob.glob(op.join(file_path, '*ica_solution.fif'))
+            raw_file = glob.glob(op.join(file_path, '*ica.fif'))
+            if ica_file:
+                ica_files.append(ica_file[0])
+            if raw_file:
+                raw_files.append(raw_file[0])
```

Comment:
> there is a lot of duplicate code in this file. It would be nice if we could make it more [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

I totally agree with you :) I'll try to put togethere the similar code, thks :)